packages/transaction-controller/src/TransactionController.ts

```diff
@@ -800,11 +802,15 @@ export class TransactionController extends BaseController<
       this.getNetworkState();
     const { chainId: currentChainId } = providerConfig;
     const newTransactions = this.state.transactions.filter(
-      ({ networkID, chainId }) => {
+      ({ networkID, chainId, transaction }) => {
         // Using fallback to networkID only when there is no chainId present. Should be removed when networkID is completely removed.
         const isCurrentNetwork =
           chainId === currentChainId ||
           (!chainId && networkID === currentNetworkID);
+        if (address) {
+          const isFromAddress = transaction.from === address;
```

Comment:
Just in case the `address` is not checksummed and the transaction `from` is for example, should we lower case both when comparing as done elsewhere in the code?

Reply:
Fixed